packages/core/src/types.ts

```diff
@@ -25,12 +25,20 @@ export type RequestPolicy =
 /** How the operation has */
 export type CacheOutcome = 'miss' | 'partial' | 'hit';
 
+/** Serialisable GraphQL primitives */
+type Primitive = PrimitiveMap | number | boolean | string | null | undefined;
+
+/** A map of primitives and primitive maps */
+interface PrimitiveMap {
+  [key: string]: Primitive | Primitive[] | PrimitiveMap;
```

Comment:
```suggestion
interface PrimitiveMap {
  __typename?: string;
  [key: string]: Primitive | Primitive[] | PrimitiveMap;
```

We can make the assumption that `__typename` is always either `undefined` or a string :tada:

Reply:
Good shout!